core/src/main/java/org/jboss/pnc/rex/core/RemoteEntityClient.java

```diff
@@ -187,4 +198,45 @@ private org.jboss.pnc.api.dto.Request getCallbackRequest(String taskName, String
         }
         return callbackRequest;
     }
+
+    /**
+     * get the task results of the direct dependencies of the task, only if the configuration allows so
+     *
+     * @param task task to return task results
+     * @return Map of taskName and the result if configuration allows sharing of results, otherwise null
+     */
+    private Map<String, Object> getTaskResults(Task task) {
+
+        if (task.getConfiguration() != null && task.getConfiguration().isPassResultsOfDependencies()) {
+
+            Map<String, Object> taskResults = new HashMap<>();
+            // tasks that this task depends on
+            Set<String> dependencies = task.getDependencies();
+
+            // add dependency results to the map
+            for (String taskName : dependencies) {
+
+                Task previousTask = taskRegistry.getTask(taskName);
+
+                if (previousTask == null) {
+                    throw new RuntimeException("Task " + taskName + " cannot be found!");
```

Comment:
Either use taskRegistry.getRequiredTask or throw TaskMissingException. If you throw just RuntimeException, then the internal fault tolerance would retry the jobs but fail in the same place as I think this is non-recoverable. TaskMissingException is one of the exceptions that are labeled as non-recoverable and retries are ignored. To see which exceptions are non-recoverable you can look at application.yaml and scheduler.options.retry-policy.abortOn key.

The reason why everything is wrapped in fault tolerance (in retries mainly) is because the scheduler is designed that transactions can fail at anytime because of concurrency so stuff needs to be retried. Also everything has to be in transation so that we have atomicity and consistent data.

Reply:
Done!